src/server.ts

```diff
@@ -41,43 +39,32 @@ export interface OperationMessage {
   type: string;
 }
 
-export type ExecuteReactiveFunction = (
-  schema: GraphQLSchema,
-  document: DocumentNode,
-  rootValue?: any,
-  contextValue?: any,
-  variableValues?: {[key: string]: any},
-  operationName?: string,
-) => IObservable<ExecutionResult>;
-
-export type ExecuteFunction = (
-  schema: GraphQLSchema,
-  document: DocumentNode,
-  rootValue?: any,
-  contextValue?: any,
-  variableValues?: {[key: string]: any},
-  operationName?: string,
-) => Promise<ExecutionResult>;
-
-export type SubscribeFunction = (
-  schema: GraphQLSchema,
-  document: DocumentNode,
-  rootValue?: any,
-  contextValue?: any,
-  variableValues?: {[key: string]: any},
-  operationName?: string,
-) => AsyncIterator<ExecutionResult>;
-
-export interface Executor {
-  execute?: ExecuteFunction;
-  executeReactive?: ExecuteReactiveFunction;
-  subscribe?: SubscribeFunction;
-}
+export type ExecuteFunction = (schema: GraphQLSchema,
+                               document: DocumentNode,
+                               rootValue?: any,
+                               contextValue?: any,
+                               variableValues?: { [key: string]: any },
+                               operationName?: string) => Promise<ExecutionResult> | AsyncIterator<ExecutionResult>;
```

Comment:
If this is attempting to model graphql-js's execute function, that always returns a Promise. Only subscribe returns an AsyncIterator